PRINCIPLES.md

```diff
@@ -0,0 +1,254 @@
+
+1. [**The principle of declarative configuration**](#1-declarative-configuration)
+
+    A system managed by GitOps must have its _Desired State_ expressed declaritively as data.
+    This declarative data must be in a format writable and readable by both humans and software.
+
+2. [**The principle of immutable configuration versions**](#2-immutable-configuration-versions)
+
+    _Desired State_ is stored in a way that supports versioning, immutability of versions, and retains a complete version history.
+    We call systems that store desired state in this way _State Stores_.
+
+3. [**The principle of continuous state reconciliation**](#3-continuous-state-reconciliation)
+
+    Software agents continuously, and automatically, compare a system's _Actual State_ to its _Desired State_.
+    If the actual and desired states differ, automated actions are immediately attempted to reconcile them.
+    These differences could be due to the actual state drifting from the desired state, or the desired state changing intentionally.
+
+4. [**The principle of operations through declaration**](#4-operations-through-declaration)
+
+    When wishing to operate on a software system, a human or software agent will not interact with the running system and modify it directly.
+    Instead, the agent will create a new declarative version of the desired state in the state store.
+
+## Introduction
+
+The software systems that we manage vary widely; from battery powered embedded systems driven by microcontrollers, to globally distributed applications with millions of users. 
+Most systems are composed of subsystems themselves made of software.
+It is impossible to comprehensively outline all the specific practices for managing such a variety of systems.
+
+However, despite the myriad differences, several important principles emerge that simplify the task of reliably managing and operating _all_ software systems at scale. GitOps attempts to capture some of these principles in a coherent framework.
+The GitOps principles can be applied to managing entire software systems or applied only to parts of larger systems.
+
+See also the [Rationale for GitOps](RATIONALE.md).
+
+## Scope
+
+GitOps concerns the verifiable behaviour of computer systems and their interfaces.
+Specifically, GitOps is _not_ about human processes, and is not intended as a model for judging human organisational designs and operational practices.
+
+The GitOps principles are to be used as guiding principles in the development of modern software and system operations. They do not form a concrete specification.
+
+The GitOps principles are a direction, not a destination. They should be applied pragmatically. For example, whilst desirable to apply them strictly to an entire software systems, they can also be applied loosely to selected parts of larger systems as part of a progressive adoption.
+
```

Comment:
I do think that GitOps also makes sense in environments that don't have a healthy balance yet.
There, GitOps can act as an agent that reconciles the mismatch. ;)

Reply:
I second this sentiment. (Good link about socio-technical systems too.) I wanted to make sure that the scope excluded human decision making explicitly. I've re-worded to reflect this and make it clearer. Do you think the meaning is clearer and more accurate now?

Reply:
I appreciated the original statement in line 83 and found it valuable/clarifying to directly callout the separation between our intent, the principles and a specification.   It also posed a small question with the inclusion of software development, whereas most GitOps discussion revolves around operations.  If I change the source code of a system to account for new business logic, does that fall under the umbrella of GitOps, which is heavily weighted on Ops?  How far left does do we envision GitOps spanning?

Reply:
@kmb385 currently this idea is captured in a few places in this PR (specifically "systems" or "subsystems"). Does this address your concern?

Here:

>By "Configuration", we mean _data that defines how a system or subsystem should behave_.
>
>For example, the same web server code may be running on thousands of different servers managed by hundreds of different companies.
>The behaviour of an individual webserver will differ based on how it is configured.
Configuration data is typically in the form of files or arguments to a computer program, but some systems may also currently use configuration databases or remote configuration services. Configuration also includes data about what version of code a software system should run, so software version information is also considered configuration.
>
>Together, the aggregate of all configuration data for a system form its "Desired State". The "Desired State" of a system is defined as data sufficient to recreate the system from nothing so that instances of the system are behaviorally indinstinguishable.

Also here:

>The definition of a _system_ can be quite broad, and may incorporate human as well as programmatic processes.
For example, is a company's sales process a system?
>Should GitOps be applied to it?
>Although a vision in which the GitOps practices are applied generally to all processes, human or otherwise is compelling, a more pragmatic approach is preferable, to avoid the risk of attempting to "boil the ocean".
>
>Instead, we should focus on subsystems where the Desired State is well defined, implement the GitOps principles there, and grow out to capture more systems from that initial subsystem. <!-- This progressive approach to GitOps is elaborated further in [the guide to adopting GitOps.md](ADOPTING_GITOPS.md). -->

The PR also retains this from the README:

>GitOps is fast becoming the methodology of choice for operating modern cloud native infrastructure and applications

Reply:
@scottrigby It does, I'm very new to the project and reading the definition of "software system" in the definitions section really helped.  My interpretation of that definition is that it does include the application source code.